tests/unit/test_featuretoggles.py

```diff
@@ -0,0 +1,100 @@
+from typing import List
+
+import pytest
+from pytest_mock import MockerFixture
+
+from ni_measurementlink_service._featuretoggles import (
+    CodeReadiness,
+    FeatureNotSupportedError,
+    FeatureToggle,
+    get_code_readiness_level,
+    requires_feature,
+)
+
+RELEASE_FEATURE = FeatureToggle("RELEASE_FEATURE", CodeReadiness.RELEASE)
+NEXT_RELEASE_FEATURE = FeatureToggle("NEXT_RELEASE_FEATURE", CodeReadiness.NEXT_RELEASE)
+INCOMPLETE_FEATURE = FeatureToggle("INCOMPLETE_FEATURE", CodeReadiness.INCOMPLETE)
+PROTOTYPE_FEATURE = FeatureToggle("PROTOTYPE_FEATURE", CodeReadiness.PROTOTYPE)
+
+
+@requires_feature(PROTOTYPE_FEATURE)
+def _prototype_function(x: int, y: str, z: List[int]) -> str:
+    return _prototype_function_impl(x, y, z)
+
+
+def _prototype_function_impl(x: int, y: str, z: List[int]) -> str:
+    return ""
+
+
+def test___default_code_readiness_level___get_code_readiness_level___equals_prototype() -> None:
+    assert get_code_readiness_level() == CodeReadiness.PROTOTYPE
+
+
+@pytest.mark.use_code_readiness(CodeReadiness.RELEASE)
+def test___use_release_readiness___get_code_readiness_level___equals_release() -> None:
+    assert get_code_readiness_level() == CodeReadiness.RELEASE
+
+
+@pytest.mark.use_code_readiness(CodeReadiness.NEXT_RELEASE)
+def test___use_next_release_readiness___get_code_readiness_level___equals_next_release() -> None:
+    assert get_code_readiness_level() == CodeReadiness.NEXT_RELEASE
+
+
+def test___default_code_readiness_level___is_enabled___returns_true() -> None:
+    assert RELEASE_FEATURE.is_enabled
+    assert NEXT_RELEASE_FEATURE.is_enabled
+    assert INCOMPLETE_FEATURE.is_enabled
+    assert PROTOTYPE_FEATURE.is_enabled
+
+
+@pytest.mark.use_code_readiness(CodeReadiness.INCOMPLETE)
+def test___use_incomplete_readiness___is_enabled___reflects_code_readiness_level() -> None:
+    assert RELEASE_FEATURE.is_enabled
+    assert NEXT_RELEASE_FEATURE.is_enabled
+    assert INCOMPLETE_FEATURE.is_enabled
+    assert not PROTOTYPE_FEATURE.is_enabled
+
```

Comment:
Did you intentionally omit a next_release test?

Reply:
These were originally spot checks but since they cover 3/4 cases I guess I'll add the 4th.